crates/proof-of-sql/src/base/database/column_repetition_operation.rs

```diff
@@ -120,19 +136,40 @@ impl RepetitionOp for ColumnRepeatOp {
     fn op<T: Clone>(column: &[T], n: usize) -> impl Iterator<Item = T> {
         repeat_slice(column, n)
     }
+
+    fn op_fixed_size_binary(col_bytes: &[u8], width: usize, n: usize) -> Vec<u8> {
+        let rows: Vec<_> = col_bytes.chunks_exact(width).collect();
+        let mut out = Vec::with_capacity(rows.len() * width * n);
+        for row in rows.iter().cycle().take(rows.len() * n) {
+            out.extend_from_slice(row);
+        }
+        out
+    }
 }
 
 pub struct ElementwiseRepeatOp {}
 impl RepetitionOp for ElementwiseRepeatOp {
     fn op<T: Clone>(column: &[T], n: usize) -> impl Iterator<Item = T> {
         repeat_elementwise(column, n)
     }
+
+    fn op_fixed_size_binary(col_bytes: &[u8], width: usize, n: usize) -> Vec<u8> {
+        let mut out = Vec::with_capacity(col_bytes.len() * n);
+        out.extend(
+            col_bytes
+                .chunks_exact(width)
+                .flat_map(|row| iter::repeat(row).take(n))
+                .flatten()
+                .copied(),
+        );
+        out
```

Comment:
Please avoid mutability
```suggestion
            col_bytes
                .chunks_exact(width)
                .flat_map(|row| iter::repeat(row).take(n))
                .flatten()
                .copied()
                .collect()
```